prac_07/convert_miles_km.py

```diff
@@ -0,0 +1,42 @@
+from kivy.app import App
+from kivy.core.window import Window
+from kivy.lang import Builder
+from kivy.properties import StringProperty
+
+MILES_IN_KM = 1.60934
+
+
+class ConvertMilesToKM(App):
+    km_output = StringProperty()
+
+    def build(self):
+        """Build app in window 500x200"""
+        self.title = "Convert Miles to KM"
+        Window.size = (500, 200)
+        self.root = Builder.load_file("convert_miles_km.kv")
+        return self.root
+
+    def validate_number(self):
+        """Validates and returns 0.0 if not a valid number"""
+        if self.root.ids.miles_input.text == '':
+            return 0.0
+        else:
+            try:
+                value = float(self.root.ids.miles_input.text)
+                return value
+            except ValueError:
+                return 0.0
+
+    def handle_increment(self, number):
+        """increment the user input number by 1 or -1 based on button selection"""
+        number_to_increment = self.validate_number()
+        self.root.ids.miles_input.text = str(number_to_increment + number)
```

Comment:
Consider adding km_conversion to function to satisfy practical requirement.

Reply:
Will commit this now, However conversions still happen on demand without this been added. So just confused as to what it would do extra?

Reply:
If I’m not wrong, they want you to convert on_text and when you increase/decrease the increment